### PR TITLE
Fixes #20742 - Make hammer host reports work

### PR DESCRIPTION
**Pull request by mbacovsky**
Added proper mapping of id and name options to search queries. Made host
parameter required. I found out there is no name in reports and fixed
the id resolver. --search was fixed. Added some tests.

### CODE REVIEW

**Comment by xprazak2** [comment]
I tested, works nicely. I will close my PR in favour of this.

File: lib/hammer_cli_foreman/host.rb [hunk 1]
```diff
@@ -213,12 +213,33 @@ def send_request
...
+        validator.any(:option_name, :option_id).required
+      end
+
+      def request_params
+        params = super
+        search = []
+        search << params['search'] if params['search']
+        
+        hostname = get_option_value('name')
+        search << "host=\"#{hostname}\"" if hostname
```
Comment (ares) [inline lib/hammer_cli_foreman/host.rb]: you could use `%Q(host="#{hostname}")` to avoid escaping if you want

**Comment by ares** [comment]
LGTM, left one minor comment

**Comment by mbacovsky** [comment]
@ares, @xprazak2, thanks, updated.

**Comment by ares** [comment]
:+1:

**Pull request merged**